spec/compiler/semantic/instance_var_spec.cr

```diff
@@ -5042,4 +5042,71 @@ describe "Semantic: instance var" do
       Second.new.hash
       )) { hash_of(hash_of(int32, int32).virtual_type, int32).virtual_type }
   end
+
+  it "doesn't solve instance var initializer in instance context (1) (#5876)" do
+    assert_error %(
+      class Foo
+        @x : Int32 = bar
+
+        def bar
+          1
+        end
+      end
+
+      Foo.new
+      ),
+      "undefined local variable or method 'bar'"
```

Comment:
can we detect this and emit a specialized "did you mean" error message? I suspect it wouldn't be easy but worth asking.

Reply:
I think it's possible, if a lookup fails, see if there's a method with that name in the class scope... but I don't feel like implement that right now (someone else can try it, though)